packages/at_chops/lib/src/at_chops_impl.dart

```diff
@@ -39,6 +39,10 @@ class AtChopsImpl extends AtChops {
     try {
       encryptionAlgorithm ??=
           _getEncryptionAlgorithm(encryptionKeyType, keyName)!;
+      if (encryptionAlgorithm is SymmetricEncryptionAlgorithm && iv == null) {
+        throw InvalidDataException(
```

Comment:
InvalidDataException is not the right exception to throw.
modify exception text - "Initialization vector is required for SymmetricKeyEncryption"

Reply:
[00b72de](https://github.com/atsign-foundation/at_libraries/pull/554/commits/00b72dea48a0b9cafd1d66a4178047b2e2444afa)